iceoryx_examples/ice_multi_publisher/ice_multi_publisher.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/popo/modern_api/publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "topic_data.hpp"
+
+#include <chrono>
+#include <csignal>
+#include <iostream>
+#include <string>
+
+bool killswitch = false;
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    killswitch = true;
+}
+
+void send(uint32_t id, const char* instanceName, std::chrono::milliseconds delay)
+{
+    iox::capro::IdString instance{iox::cxx::TruncateToCapacity, instanceName};
+
+    iox::popo::TypedPublisher<CounterTopic> publisher({"CounterTopic", instance, "Counter"});
+    publisher.offer();
+
+    for (uint32_t counter = 0U; !killswitch; ++counter)
+    {
+        CounterTopic data{counter, id};
+        publisher.publishCopyOf(data);
+
+        // prevent undesired output interleaves of independent sender threads
+        std::stringstream s;
+        s << instance.c_str() << " sending: " << data << std::endl;
+        std::cout << s.str();
+
+        std::this_thread::sleep_for(delay);
+    }
+
+    publisher.stopOffer();
+}
+
+int main()
+{
+    signal(SIGINT, sigHandler);
+
+    iox::runtime::PoshRuntime::initRuntime("/iox-publisher");
+
+    // generates multiple publishers which send the same topic (with different or same instances)
```

Comment:
This naming will be confusing to the ROS community I guess. But yes, we can define that topics are the same if and only if all 3 match. I will change the example but then there is no real need to have a third publisher (since it would publish on a different topic which we already demonstrate elsewhere).

Reply:
In the rmw_iceoryx we currently only use two of the three strings. One for the topic name with namespace and one for the type. For now using all three together for uniquely identifying an iceoryx topic is fine. Other variants need some discussions first

Reply:
only two publishers with the same identifiers remain.